src/sentry/eventstore/models.py

```diff
@@ -681,6 +682,43 @@ def for_group(self, group: Group):
         return GroupEvent.from_event(self, group)
 
 
+@dataclass
+class IssueEvidence:
+    name: str
+    value: str
+    # Feels like we need more attributes to allow us to decide which things to include in verbose and
+    # condensed displays. We'll use all the evidence in issue details and emails, but only important
+    # rows in slack, for example. Some thoughts:
+    # important: bool
+    # include_in: Sequence[str] # List of places this should be included in, like UI, email, slack.
+    # Maybe too configurable? Also doesn't expand well as we add more integration types. Could be
+    # more general like: UI, alerts_email, alerts_messaging, integration_ticketing, etc.
+
+
+@dataclass
+class IssueOccurrence:
+    """
+    A class representing a specific occurrence of an issue. Separate to an `Event`. An `Event` may
+    have 0-M `IssueOccurrences` associated with it, and each `IssueOccurrence` is associated with
+    one `Event`.
+
+    Longer term, we will change this relationship so that each `IssueOccurrence` is the primary row
+    and is passed around, and has an optional `Event` associated with it.
+    """
+
+    id: str
+    event_id: str  # Maybe optional? Although I think that we should create an event for all issue
+    # categories, even if the backing data type doesn't explicitly have Events
+    fingerprint: Sequence[str]
+    issue_title: str
+    # subtitle: str # We haven't defined what this will look like just yet
+    resource_id: str
```

Comment:
Is this the external id for referencing external systems? Maybe this should be optional?

Reply:
I was thinking that this could be an event id for things like errors and transactions, and a generic other id for external systems. I guess there's no reason we have to require it though...

Reply:
Hmm, but we have `event_id` up above? I was thinking `event_id` is always present even for external systems, and `resource_id` is the "other id" that references entities we don't manage.

Reply:
Yeah actually you're right, I confused myself here but it makes sense that event_id is specific to our internal events. So in that case, I think you're right that it can be optional.